src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -534,52 +529,47 @@ public with sharing class PMT_PaymentWizard_CTRL {
 
     }
 
-    /*******************************************************************************************************
-    * @description Displays an error message when then user has no permission to modify payments
-    * @return void
-    */
-    private void displayNoAccessMessage() {
-        String errorMsg;
-
-        if (isWriteoffWizard == true) {
-            errorMsg = System.Label.pmtWizardWriteoffNoAccessMessage;
-        } else if (isScheduleWizard == true) {
-            errorMsg = System.Label.pmtWizardPaymentsNoAccessMessage;
-        } else {
-            errorMsg = System.Label.pmtWizardMsgNoOppFound;
-        }
-
-        ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, errorMsg));
-    }
-
     /*******************************************************************************************************
     * @description Returns true if the user has permissions to access, create and delete npe01__oppPayment__c
     * and all relevant fields are both readable and modifiable
     * @return Boolean
     */
-    private Boolean hasPaymentAccess() {
-        Boolean accessResult = false;
-
-        SObjectType pmt = npe01__OppPayment__c.getSObjectType();
-        if (perms.canCreate(pmt) && UTIL_Permissions.canDelete(Schema.SObjectType.npe01__oppPayment__c.getName(), false)) {
-            Set<SObjectField> sObjectFields = new Set<SObjectField>{
-                npe01__oppPayment__c.fields.npe01__check_reference_number__c,
-                npe01__oppPayment__c.fields.npe01__paid__c,
-                npe01__oppPayment__c.fields.npe01__payment_amount__c,
-                npe01__oppPayment__c.fields.npe01__payment_date__c,
-                npe01__oppPayment__c.fields.npe01__payment_method__c,
-                npe01__oppPayment__c.fields.npe01__scheduled_date__c,
-                npe01__oppPayment__c.fields.npe01__written_off__c
-            };
-            if (!(perms.canRead(pmt, sObjectFields) && perms.canUpdate(pmt, sObjectFields))) {
-                return accessResult;
-            }
-            accessResult = true;
+    private Boolean hasFeatureAccess() {
+        SObjectType payment = npe01__OppPayment__c.getSObjectType();
+        SObjectType opportunity = Opportunity.getSObjectType();
+        if (canCheckPermissionsFor(payment)) {
```

Comment:
Just reading through this, I'm not sure if this "canCheckPermissionsFor" is clear in its intent in the naming. I see what you're going for in checking for create/delete first before checking for read/update, but for the sake of readability here I'm wondering if it would be better to just have the "hasAccessTo" check without the additional "create" / "delete" check and we could roll the "create" and "delete" check into the "hasAccessTo" method which would then comprise all of the required checks.

Reply:
I agree with your comment here, I'll move the call to `canCheckPermissionsFor` into `hasAccessTo`.

Reply:
Updated!